### PR TITLE
storageFormats: Add variable to store auxiliary values in eventMetadata

### PR DESCRIPTION
**Pull request by stefan-wallner**

### CODE REVIEW

File: storageFormats/eventMetadata.h [hunk 1]
```diff
@@ -40,6 +40,10 @@ namespace rpwa {
 		const std::vector<std::string>& productionKinematicsParticleNames() const { return _productionKinematicsParticleNames; }
 		const std::vector<std::string>& decayKinematicsParticleNames() const { return _decayKinematicsParticleNames; }
 		const std::vector<std::string>& additionalSavedVariableLables() const { return _additionalSavedVariableLabels; }
+		const std::map<std::string,double>& additionalSavedValues() const { return _additionalSavedValues; }
+		double getAdditionalSavedValue(const std::string& name) const { return _additionalSavedValues.at(name); }
+		void setAdditionalSavedValue(const std::string& name, const double value) { _additionalSavedValues[name] = value; }
+		bool hasAdditionalSavedValue(const std::string& name) const { return _additionalSavedValues.find(name) != _additionalSavedValues.end(); }
```
Comment (suhlatwork) [inline storageFormats/eventMetadata.h]: you could shorten this to `_additionalSavedValues.count(name) > 0`.

File: storageFormats/eventMetadata.cc [hunk 2]
```diff
@@ -316,6 +316,22 @@ eventMetadata* rpwa::eventMetadata::merge(const vector<const eventMetadata*>& in
...
 		mergee->setBinningMap(mergedBinningMap);
 	}
+
+	for (const auto& input : inputData) {
+		for (const auto& nameValue : input->additionalSavedValues()) {
+			if (not mergee->hasAdditionalSavedValue(nameValue.first)) {
+				mergee->setAdditionalSavedValue(nameValue.first, nameValue.second);
+			} else { // the additional value exists in multiple files -> only merge if it is the same
+				if (mergee->getAdditionalSavedValue(nameValue.first) != nameValue.second) {
+					printWarn<< "additional value '" << nameValue.first << "' has different values in the different files." << endl;
```
Comment (suhlatwork) [inline storageFormats/eventMetadata.cc]: please add a space between `printWarn` and `<<`.

Comment (suhlatwork) [inline storageFormats/eventMetadata.cc]: also `printWarn` should be `printErr` as this errors appears fatal to me.

Comment (suhlatwork) [inline storageFormats/eventMetadata.cc]: but apparently we never use `printErr` in this file, so never mind.

File: storageFormats/eventMetadata.cc [hunk 3]
```diff
@@ -316,6 +316,22 @@ eventMetadata* rpwa::eventMetadata::merge(const vector<const eventMetadata*>& in
 	if(mergeDiffMeta) {
 		mergee->setBinningMap(mergedBinningMap);
 	}
+
+	for (const auto& input : inputData) {
```
Comment (stefan-wallner) [inline storageFormats/eventMetadata.cc]: Thanks a lot for your comment.
I fully agree that one should be cautious with using `auto`, but in the context of for loops I thinks it is very useful, because you usually spare lots of characters, which makes the code much more readable.

File: storageFormats/eventMetadata.h (hunk 1 above)
Comment (stefan-wallner) [inline storageFormats/eventMetadata.h]: This would be shorter, but I think the semantic for find is more clear, because here we want to find whether the map has that key, not count it. 
Performance wise it does not matter.

File: pyInterface/bindings/storageFormats/eventMetadata_py.cc [hunk 4]
```diff
@@ -86,6 +86,9 @@ void rpwa::py::exportEventMetadata() {
 		.def("productionKinematicsParticleNames", &eventMetadata_productionKinematicsParticleNames)
 		.def("decayKinematicsParticleNames", &eventMetadata_decayKinematicsParticleNames)
 		.def("additionalSavedVariableLables", &eventMetadata_additionalSavedVariableLables)
```
Comment (bgrube) [inline pyInterface/bindings/storageFormats/eventMetadata_py.cc]: Not your fault, but it would be nice to fix typo: "additionalSavedVariableLables" -> "additionalSavedVariableLabels"

File: pyInterface/bindings/storageFormats/eventMetadata_py.cc [hunk 5]
```diff
@@ -86,6 +86,9 @@ void rpwa::py::exportEventMetadata() {
...
+		.def("getAdditionalSavedValue", &rpwa::eventMetadata::getAdditionalSavedValue)
```
Comment (bgrube) [inline pyInterface/bindings/storageFormats/eventMetadata_py.cc]: In ROOTPWA we usually not decorate getters with "get", so it would be nice to rename "getAdditionalSavedValue" -> "additionalSavedValue"

File: storageFormats/eventMetadata.cc [hunk 6]
```diff
@@ -316,6 +316,22 @@ eventMetadata* rpwa::eventMetadata::merge(const vector<const eventMetadata*>& in
 	if(mergeDiffMeta) {
 		mergee->setMultibinBoundaries(mergedmultibinBoundaries);
 	}
+
+	for (const auto& input : inputData) {
```
Comment (bgrube) [inline storageFormats/eventMetadata.cc]: Why make a separate loop here? Why not include it in the loop over the input data above?

**Comment by suhlatwork** [comment]
Updated with respect to 40e7594900237153d7b9780c2152e7b3f3d50fb7: replaced more occurrences of `userString` with `auxString`, style.

**Comment by suhlatwork** [comment]
Merged, thank you!
